x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_search_bar/analytics_search_bar.tsx

```diff
@@ -20,6 +20,68 @@ import {
   Value,
   DataFrameAnalyticsListRow,
 } from '../analytics_list/common';
+import { ModelItem } from '../models_management/models_list';
+
+export function filterAnalyticsModels(
```

Comment:
I find this function a bit confusing. We have a collection that we want to filter based on the set of clauses.
Why not have something like 
```typescript
return items.filter(item => {
 return // check item against clauses
)
```
?

Reply:
🤔  Agree the filtering functionality could be improved. Currently it matches what's been used in the anomaly detection list and analytics list. Might be worth refactoring all of them so the search always works the same way. I think that change is outside the scope of this PR.

Reply:
I'd understand if you reused `filterJobs` from `x-pack/plugins/ml/public/application/jobs/jobs_list/components/utils.js`, but you introduced a brand new function. It would be great to have a generic one that allows you to filter the collection based on clauses. I suspect we don't need to invent our own, just get an existing funtion from EUI.